dds/DCPS/transport/rtps_udp/RtpsUdpTransport.cpp

```diff
@@ -244,32 +244,54 @@ RtpsUdpTransport::use_datalink(const RepoId& local_id,
   }
 }
 
-ACE_INET_Addr
-RtpsUdpTransport::get_connection_addr(const TransportBLOB& remote,
-                                      bool* requires_inline_qos,
-                                      unsigned int* blob_bytes_read) const
+std::pair<ACE_INET_Addr, ACE_INET_Addr>
+RtpsUdpTransport::get_connection_addrs(const TransportBLOB& remote,
+                                       bool* requires_inline_qos,
+                                       unsigned int* blob_bytes_read) const
 {
   using namespace OpenDDS::RTPS;
   LocatorSeq locators;
   DDS::ReturnCode_t result =
     blob_to_locators(remote, locators, requires_inline_qos, blob_bytes_read);
   if (result != DDS::RETCODE_OK) {
-    return ACE_INET_Addr();
+    return std::make_pair(ACE_INET_Addr(), ACE_INET_Addr());
   }
 
+  ACE_INET_Addr uc_addr;
+  ACE_INET_Addr mc_addr;
+  bool uc_addr_valid = false;
+  bool mc_addr_valid = false;
   for (CORBA::ULong i = 0; i < locators.length(); ++i) {
     ACE_INET_Addr addr;
     // If conversion was successful
     if (locator_to_address(addr, locators[i], false) == 0) {
       // if this is a unicast address, or if we are allowing multicast
-      if (!addr.is_multicast() || config().use_multicast_) {
-        return addr;
+      if (!addr.is_multicast()) {
```

Comment:
```suggestion
      if (addr.is_multicast()) {
```
and flip the two then/else blocks